### ISSUE TITLE
postgres support

### ISSUE THREAD

**Issue by huobazi** [flarum/core#290]
ref: http://discuss.flarum.org/d/597-does-flarum-support-postgres

**Comment by dz0ny** [comment]
:+1:

**Comment by jirutka** [comment]
+1

**Comment by korya** [comment]
It would be great to see postgres officially supported.

**Comment by vjousse** [comment]
Just FYI, I'm having a look at the psql support. If you have any news on your side, don't hesitate to let me know.

**Comment by fdutheil** [comment]
I'd like to add a +1, but I'm not sure of the status of this feature request: is it already upstream and needs testing or is it cancelled after the WIP has stalled?

**Comment by vladipus** [comment]
I too would like to know if the pg support is there.

**Comment by vjousse** [comment]
I did some significant work here: https://github.com/flarum/core/pull/856 but was not able to finish it alone.
I suppose it's not a priority right now, but as far as I remember the modifications I made allowed the forum to be installed using postgres.

**Comment by evergist** [comment]
+1

**Comment by phonicmouse** [comment]
+1

**Comment by phonicmouse** [comment]
Guys can you clarify something?
I know from [here](https://discuss.flarum.org/d/597-does-flarum-support-postgres) that you use Laravel's drivers, which support PostgreSQL. So if i try to install flarum simply using PostgreSQL instead of MySQL it just works or not?

**Comment by franzliedke** [comment]
@phonicmouse As mentioned in that forum thread, the driver needs to be made available in the installer. Once that is done, any problems that arise have to be fixed. Somebody needs to experiment with it and see how much fails. Do you want to give it a shot? :grinning:

**Comment by luceos** [comment]
From the top of my head, there are some incompatibility issues with either some (Gambit) queries and/or migrations. Either of these are specifically written against a MySQL database and need to be modified.

**Comment by phonicmouse** [comment]
@franzliedke I'll definitely do it. As soon as i have some spare time I'll work on it 😋 
@luceos Well, after we locate them, we just have to make them cross-database, which shouldn't be too hard, but just a bit long, but i've never looked at the code, so...😭

**Comment by mkarnicki** [comment]
@phonicmouse 

>  we just have to make them cross-database, which shouldn't be too hard

Note there was already an attempt to support Postgres (see above in the thread) and it didn't turn out so simple. Mostly because the integration has to be correct in all aspects from the Flarum perspective, so: easy for the users to choose/setup. In any case, I wish you the very best if you take this on :)! Quite a few folks are looking forward to Postgres support. Few of us have time to work on it though.

**Comment by phonicmouse** [comment]
@mkarnicki Yeah, I know, I will definitely have a look at that. Yeah I think the main issue is just finding the time to work on this. As soon as I have some spare time I'll try to solve this.

**Comment by fr0der1c** [comment]
It has been for 4 years...

**Comment by mkarnicki** [comment]
I still think this would be a valuable addition, esp. that Laravel makes it so easy to interchange database types one would use, so I'm leaving this comment here hoping the report is not killed by the stale bot and that it maybe receives some attention in the future. Cheers.

**Comment by jtojnar** [comment]
usntale

**Comment by franzliedke** [comment]
After some discussion in the core team (a while ago)...

We changed our opinion and currently do not plan to tackle this as part of our roadmap. We will, of course, be open for / discuss community contributions for this feature - it was brought up from the community in the first place anyway.

So I am closing this issue to avoid appearing like we plan to work on this feature.

**Comment by luceos** [comment]
Even though this saddens me deeply. I think the motto "do one thing and do it well" is key here. Let's focus on getting Flarum working exceptionally well with MySQL.

**Comment by c33s** [comment]
i ask myself why so many people use database abstraction layers like doctrine or many parts from symfony and still develop mysql only code.

**Comment by tankerkiller125** [comment]
@c33s Often because those libraries and abstractions make working with databases much easier, but often very advanced things (for example search) requires using raw SQL which often is database specific. And since ALL host offer MySQL (including shared host) MySQL tends to win the argument for web apps looking for the largest use. Other databases are great for specific kinds of things (for example Gitlab has to use Postgres because of the database features it needs) but for something like Flarum it makes the most sense to focus on MySQL.

**Comment by luceos** [comment]
I'd like to point out that we are not entirely excluding alternative Eloquent-compatible database drivers for future versions. But we do for now to prevent any delays on releasing stable.

**Comment by blackandred** [comment]
I personally don't see any sense in using MySQL in web applications, recently I moved from MySQL to PostgreSQL in a web application - the cost of maintaining PostgreSQL is much lower in my opinion.

**Comment by askvortsov1** [comment]
> I personally don't see any sense in using MySQL in web applications, recently I moved from MySQL to PostgreSQL in a web application - the cost of maintaining PostgreSQL is much lower in my opinion.
> 
> It would be nice to see Flarum on PostgreSQL so I should not maintain MySQL instance. Apps in other languages uses more PostgreSQL than MySQL, so it is natural that a PostgreSQL instance in most cases could be shared between applications.

I agree with all of this. There are many advantages of PostgreSQL. One of the ones I came across recently is the ability to put structural changes in transactions: that could have been quite helpful in #2258.

Even if it's far in the future, I think this issue is worth considering. I've already kept it in mind when approaching some other issues (#2258, #2612, etc).

**Comment by originswift-sys** [comment]
> I personally don't see any sense in using MySQL in web applications, recently I moved from MySQL to PostgreSQL in a web application - the cost of maintaining PostgreSQL is much lower in my opinion.
> 
> It would be nice to see Flarum on PostgreSQL so I should not maintain MySQL instance. Apps in other languages uses more PostgreSQL than MySQL, so it is natural that a PostgreSQL instance in most cases could be shared between applications.

What he said, basically.

**Comment by mkarnicki** [comment]
Basically, there are 18 subscribers getting notified of every comment on this issue, so the best way to show your support for issues on GitHub is to upvote the original issue at the top by @huobazi . Thanks, have a good day.
